### PR TITLE
post operation on /apis for api deploy

### PR DESCRIPTION
**Pull request by topliceanurazvan**
## Changes

- Adds POST operation on `/apis` for the API deploy

### CODE REVIEW

File: api/openapi.yaml [hunk 1]
```diff
@@ -56,6 +56,27 @@ paths:
...
+      description: "Deploys a new API to the cluster"
+      operationId: deployApi
+      requestBody:
+        description: "API content that needs to be deployed"
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
```
Comment (olensmar) [inline api/openapi.yaml:72]: are properties necessary? I think it could be just

```
schema:
   type: object
```

File: api/openapi.yaml [hunk 2]
```diff
@@ -56,6 +56,27 @@ paths:
...
+                api:
+                  type: object
+              required:
+                - api
+      responses:
```
Comment (olensmar) [inline api/openapi.yaml:77]: add an error response for if the API is "malformed" - i.e. 
- isn't an OpenAPI 3.x definition
- doesn't contain the required x-kusk extensions for it to be deployable

File: api/openapi.yaml [hunk 3]
```diff
@@ -56,6 +56,34 @@ paths:
...
+              type: object
+      responses:
+        201:
+          description: "API deployed"
+        400:
+          description: "The content of the API is malformed"
+          content:
+            application/json:
+              schema:
+                type: object
```
Comment (olensmar) [inline api/openapi.yaml:80]: maybe 

`type: string` 

is enough to start with? @jasmingacic what do you think?

Comment (jasmingacic) [reply api/openapi.yaml:80]: It should be enough

**Review by jasmingacic — APPROVED** [review]

**Review by olensmar — COMMENTED** [review]
thanks!

**Pull request merged**